js/ui/iconGridLayout.js

```diff
@@ -379,6 +379,77 @@ const IconGridLayout = new Lang.Class({
                 Lang.bind(this, this._enumerateDirectoryFiles));
     },
 
+    _createFolderFile: function(name) {
+        let keyFile = new GLib.KeyFile();
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_NAME, name);
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_TYPE, GLib.KEY_FILE_DESKTOP_TYPE_DIRECTORY);
+
+        let dir = Gio.File.new_for_path(GLib.get_user_data_dir() + '/desktop-directories');
+        if (!dir.query_exists(null))
```

Comment:
Instead of doing this, simply call `make_directory_with_parents()` and catch the error that matches `G_IO_ERROR_EXISTS` code